### PR TITLE
Remove deprecated service-defaults upstream behavior.

### PR DESCRIPTION
**Pull request by hashi-derek**
In 1.14 the service-defaults config entry lacked the `peer` field on upstream overrides and applied the override config purely based on the `name` field matching. For 1.15, we introduced the new `peer` field and also maintained the existing behavior. If `peer` was not specified anywhere in the config entry, we assume it was not peering aware and fell back to the compatibility mode. https://developer.hashicorp.com/consul/docs/upgrading/upgrade-specific#service-defaults-overrides-for-upstream-peered-services

This commit removes that deprecated backwards-compatibility behavior. Now in 1.16, it is necessary to specify the `peer` field in order for upstream overrides to apply to a peer upstream.

### CODE REVIEW

File: website/content/docs/upgrading/upgrade-specific.mdx [hunk 1]
```diff
@@ -14,6 +14,16 @@ provided for their upgrades as a result of new features or changed behavior.
...
 
+## Consul 1.16.x
+
+#### Remove deprecated service-defaults peer upstream override behavior
+
+In Consul 1.16.x, `service-defaults` upstream [`overrides`](/consul/docs/connect/config-entries/service-defaults#overrides)
+will not apply to peer upstreams, unless the [`peer`](/consul/docs/connect/config-entries/service-defaults#peer) field is explicitly provided.
+
+Consul 1.15.x introduced a backward-compatibility behavior to allow for a smoother transition during the upgrade. This behavior will be removed in Consul 1.16.x.
+Visit the [upgrade instructions for 1.15.x](#service-defaults-overrides-for-upstream-peered-services) for more info.
```
Comment (trujillo-adam) [inline website/content/docs/upgrading/upgrade-specific.mdx:25]:
```suggestion
The `service-defaults` upstream [`overrides`](/consul/docs/connect/config-entries/service-defaults#overrides)
parameter does not apply to peer upstreams unless the [`peer`](/consul/docs/connect/config-entries/service-defaults#peer) field is explicitly provided.

This change removes the backward-compatibility behavior introduced in Consul 1.15.x. Refer to the [upgrade instructions for 1.15.x](#service-defaults-overrides-for-upstream-peered-services) for more information.
```
Switch to present tense per the style guide. Will there be a separate PR for updating the configuration entry reference?

**Review by trujillo-adam — COMMENTED** [review]
This was merged mid-review

File: website/content/docs/upgrading/upgrade-specific.mdx (hunk 1 above)
Comment (hashi-derek) [reply website/content/docs/upgrading/upgrade-specific.mdx:25]: Sorry about that. Here's the docs PR. https://github.com/hashicorp/consul/pull/16960/files
